docs/DirectManipulation.md

```diff
@@ -0,0 +1,224 @@
+---
+id: direct-manipulation
+title: Direct Manipulation
+layout: docs
+category: Guides
+permalink: docs/direct-manipulation.html
+next: linking-libraries
+---
+
+It is sometimes necessary to make changes directly to a component
+without using state/props to trigger a re-render. When using React in
+the browser for example, you sometimes need to directly modify a DOM
+node. This could be necessary for a variety of reasons, such as
+performance (avoid a render pass, useful with animations) or to simplify
+the API of a component.  Both of these apply to `TouchableOpacity`.
```

Comment:
setNativeProps is best used for continuous animations, to avoid the overhead of rendering the component hierarchy and reconciling many views. I believe we want to discourage it from an API perspective since it is imperative and stores state in the native layer (DOM, UIView, etc.)